src/geometric_algebra/generic/operators.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2020 lean-ga Team. All rights reserved.
+Released under MIT license as described in the file LICENSE.
+Authors: Utensil Song
+
+This file defines operators with local notations and no axioms.
+-/
+import tactic.lint
+
+universes u
+
+namespace geometric_algebra
+
+class has_dot               (α : Type u) := (dot : α → α → α)
+class has_wedge             (α : Type u) := (wedge : α → α → α)
+
+-- class has_geom_prod         (α : Type u) := (geom_prod : α → α → α)
+
+class has_scalar_prod       (α : Type u) := (scalar_prod : α → α → α)
+
+class has_symm_prod         (α : Type u) := (symm_prod : α → α → α)
+class has_asymm_prod        (α : Type u) := (asymm_prod : α → α → α)
+
+class has_left_contract     (α : Type u) := (left_contract : α → α → α)
+class has_right_contract    (α : Type u) := (right_contract : α → α → α)
+
+-- export has_dot (dot)
+-- export has_wedge (wedge)
+
+local infix ⬝ := has_dot.dot
+local infix ∧ := has_wedge.wedge
```

Comment:
A serious problem is that wedge is globally reserved to `infixr` which has right associativity and conflicts with our wedge. `local` only evaded the problem temporarily.

Reply:
Can you elaborate on why right-associativity is an issue?

Reply:
Also it conflicts with logical AND